### PR TITLE
PDFViewer+LibPDF: Bug fixes and quality of life improvements

### PR DESCRIPTION
**Pull request by metalvoidzz**
Continuing my work on PDF support, I came across lots of issues that I would like to ~~partially~~ fix with this PR.

In LibPDF, these mainly have to do with filters, which still need lots of work. I also made some small changes to PDFViewer to improve error handling a bit.

This is part of a larger effort to support more complex documents, I already have some exciting new features queued up that very much depend on these changes.

### CODE REVIEW

File: Userland/Libraries/LibPDF/Parser.cpp [hunk 1]
```diff
@@ -214,6 +214,7 @@ PDFErrorOr<NonnullRefPtr<NameObject>> Parser::parse_name()
             break;
 
         if (m_reader.matches('#')) {
+            m_reader.consume();
```
Comment (mattco98) [inline Userland/Libraries/LibPDF/Parser.cpp:215]:
```suggestion
        if (m_reader.consume('#')) {
```

**Review by mattco98 — APPROVED** [review]
LGTM, just one very small nitpick

File: Userland/Libraries/LibPDF/Parser.cpp (hunk 1 above)
Comment (metalvoidzz) [reply Userland/Libraries/LibPDF/Parser.cpp:215]: I don't think this is right. `consume('#')` will always consume a character, even if it doesn't end up matching `#`, which means that the name would be missing that character.

Comment (mattco98) [reply Userland/Libraries/LibPDF/Parser.cpp:215]: Oh, that's a weird API lol. nvm then

**Pull request merged**